students/eowyn/session05/mailroom.py

```diff
@@ -0,0 +1,161 @@
+#!/usr/bin/env python
+
+import sys
+
+''' global variables'''
+donor_names = ["Margaret Atwood", "Fred Armisen",
+               "Heinz the Baron Krauss von Espy"]
+donations = [[300, 555], [240, 422, 1000], [1500, 2300]]
+DONORS = dict(zip(donor_names, donations))
+
+
+
+def safe_input():
+    return None
+
+def quit_code():
+    sys.exit()
+
+def return_to_menu():
+    ''' Raise a zero division error to trigger exit out of loop'''
+    return 5/0
+
+def list_donors():
+    ''' List all DONORS '''
+    print("All Donors:")
+    [print(x) for x in DONORS]
+
+
+def generate_table():
+    ''' Pretty-print a table of donor statistics '''
+    [name, total, number, ave] = [[], [], [], []]
+    for x in DONORS:  # loop over the keys which are donor names
+        name.append(x)
+        total.append(round(sum(DONORS[x])))
+        number.append(len(DONORS[x]))
+        ave.append(round(total[-1] / number[-1]))
+    report_data = list(zip(name, total, number, ave))
+    report_data = sorted(
+        report_data, key=lambda y: int(y[1]), reverse=True)
+    print_table(report_data)
+
+
+def send_letters():
+    ''' Send thank you notes to everyone in the DONORS dict'''
+    for donor in DONORS:
+        outfile = donor.replace(" ", "_") + '.txt'
+        with open(outfile, 'w') as f:
+            f.write(generate_letter(donor))
+    print("Successfully saved letters for each donor.")
+
+
+def generate_letter(donor_name):
+    '''
+    Generate a formatted thank you note to a donor and return
+    the string of the thank you note for printing or writing.
+    '''
+    fs = "Thank you, {}, for your generosity and recent gift of ${}."
+    return fs.format(donor_name, DONORS[donor_name][-1])
+
+
+def remove_inputquotes(a_string):
+    '''Remove auxillary quotes to cleanse a_string'''
+    a_string.replace('"', '')
+    a_string.replace('"', '')
```

Comment:
why call it twice? it will replace all occurances with one call.

and this isn't doing what you want -- strings are immutable, so it's returning a new string with the replacement, but then throwing it away. you want:

a_string = a_string.replace('"', '')

and you may want to call strip() as well to get rid of extra whiltespace...